### PR TITLE
Spark as an OSGi bundle

### PR DESCRIPTION
**Pull request by upul**

### CODE REVIEW

**Comment by nirmal070125** [comment]
Please note that this PR adds following 3 orbit bundles which is used in WSO2 ML.
* Spark
* Json
* Netty

File: netty/4.0.23.wso2v1/pom.xml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>2.4.0</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            io.netty.*
```
Comment (Kishanthan) [inline netty/4.0.23.wso2v1/pom.xml]: Export packages should have a version

File: json/3.0.0.wso2v1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>2.4.0</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.json.*;version="${json.version}"
```
Comment (Kishanthan) [inline json/3.0.0.wso2v1/pom.xml]: Wrong version is exported here

File: spark/1.1.0.wso2v1/pom.xml [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+                <version>2.4.0</version>
+                <extensions>true</extensions>
+                <configuration>
+                    <instructions>
+                        <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
+                        <Bundle-Name>${project.artifactId}</Bundle-Name>
+                        <Export-Package>
+                            org.apache.spark.*,
```
Comment (Kishanthan) [inline spark/1.1.0.wso2v1/pom.xml]: Export package should have a version

File: json/3.0.0.wso2v1/pom.xml [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+                        </Export-Package>
+                        <Private-Package>
+                            org.json.*
```
Comment (Kishanthan) [inline json/3.0.0.wso2v1/pom.xml]: If you have both Export and Private package instructions on the same packages, the export will take precedence. Meaning, the above will not have any effect on not to export certain packages. What you can do is identify what are the exact packages needed to be exported and make other packages private.

File: spark/1.1.0.wso2v1/pom.xml [hunk 5]
```diff
@@ -0,0 +1,368 @@
...
+                            com.ning.compress.*,
+                            org.j_paine.*,
+                            com.codahale.metrics.*,
+                            com.fasterxml.jackson.*,
+                            org.apache.curator.*,
+                            org.apache.derby.*,
+                            org.apache.commons.lang3.*,
+                            org.xerial.snappy.*,
+                            org.objectweb.asm.*,
+                            *;resolution:=optional
```
Comment (Kishanthan) [inline spark/1.1.0.wso2v1/pom.xml]: Import packages should have a version range. When importing packages the best practice is to use the semantic version approach with pattern like com.foo.bar.*;version=”[1.0.0, 1.1.0)”,
This informs that my bundle requires com.foo.bar.* package that is exported in the version range of >=1.0.0 and <1.1.0. This is normal way to ensure that there will be no API changes of the package from a minor version.

**Comment by nirmal070125** [comment]
A new PR was issued for this #57

**Pull request closed without merging**